src/components/app/details/triggerView/ApprovalNode/ApprovalMaterial.tsx

```diff
@@ -0,0 +1,512 @@
+                <div className="security-tab-empty">
+                    <p className="security-tab-empty__title">Scan is Disabled</p>
+                </div>
+            )
+        } else if (mat.vulnerabilitiesLoading) {
+            return (
+                <div className="security-tab-empty">
+                    <Progressing />
+                </div>
+            )
+        } else if (!mat.vulnerabilitiesLoading && mat.vulnerabilities.length === 0) {
+            return (
+                <div className="security-tab-empty">
+                    <p className="security-tab-empty__title">No vulnerabilities Found</p>
+                    <p className="security-tab-empty__subtitle">{mat.lastExecution}</p>
+                </div>
+            )
+        } else
+            return (
+                <div className="security-tab">
+                    <p className="security-tab__last-scanned">Scanned on {mat.lastExecution} </p>
+                    <ScanVulnerabilitiesTable vulnerabilities={mat.vulnerabilities} />
+                </div>
+            )
+    }
+    return (
+        <>
+            {material.map((mat) => {
+                let isMaterialInfoAvailable = true
+                for (const materialInfo of mat.materialInfo) {
+                    isMaterialInfoAvailable =
+                        isMaterialInfoAvailable &&
+                        !!(
+                            materialInfo.webhookData ||
+                            materialInfo.author ||
+                            materialInfo.message ||
+                            materialInfo.modifiedTime ||
+                            materialInfo.revision
+                        )
+                    if (!isMaterialInfoAvailable) break
+                }
+                return (
+                    <div key={`material-history-${mat.index}`} className="material-history material-history--cd">
+                        {renderSequentialCDCardTitle(mat)}
+                        <div
+                            className={`material-history__top mh-66 ${
+                                !isSecurityModuleInstalled && mat.showSourceInfo ? 'dc__border-bottom' : ''
+                            }`}
+                            style={{
+                                cursor: `${mat.vulnerable ? 'not-allowed' : mat.isSelected ? 'default' : 'pointer'}`,
+                            }}
+                        >
+                            {renderMaterialInfo(mat)}
+                        </div>
+                        {mat.showSourceInfo && (
+                            <>
+                                {isSecurityModuleInstalled && (
+                                    <ul className="tab-list tab-list--vulnerability">
+                                        <li className="tab-list__tab">
+                                            <button
+                                                type="button"
+                                                onClick={(e) => {
```

Comment:
can we extract out this to separate function?

Reply:
Done